From 931c5ddaaa643bd930d752f22bc138d2e7863be4 Mon Sep 17 00:00:00 2001
From: fundamental <mark.d.mccurry@gmail.com>
Date: Tue, 9 Feb 2016 09:55:34 -0500
Subject: [PATCH 0849/1936] CMake: Add PluginLibDir Option

There's some disagreement on whether LV2 should go in
/usr/lib or /usr/lib64 on 64 bit systems.
To make it easier on packagers, there's now a flag to set the directory
which defaults to 'lib'
---
 CMakeLists.txt                          |  2 +-
 src/CMakeLists.txt                      |  2 ++
 src/Plugin/AlienWah/CMakeLists.txt      |  6 +++---
 src/Plugin/Chorus/CMakeLists.txt        |  6 +++---
 src/Plugin/Distortion/CMakeLists.txt    |  6 +++---
 src/Plugin/DynamicFilter/CMakeLists.txt |  6 +++---
 src/Plugin/Echo/CMakeLists.txt          |  6 +++---
 src/Plugin/Phaser/CMakeLists.txt        |  6 +++---
 src/Plugin/Reverb/CMakeLists.txt        |  6 +++---
 src/Plugin/ZynAddSubFX/CMakeLists.txt   | 10 +++++-----
 10 files changed, 29 insertions(+), 27 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 392e9fa4..6373c1e8 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -40,4 +40,4 @@ install(DIRECTORY instruments/banks
 install(DIRECTORY instruments/examples
     DESTINATION share/zynaddsubfx)
 install(DIRECTORY instruments/ZynAddSubFX.lv2presets
-    DESTINATION lib/lv2)
+    DESTINATION ${PluginLibDir}/lv2)
diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index 1dfcf693..c7c0f5d2 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -82,6 +82,8 @@ SET (LibloEnable ${LIBLO_FOUND} CACHE BOOL
     "Enable Liblo")
 SET (NoNeonPlease False CACHE BOOL
     "Workaround For Broken Neon Detection")
+SET (PluginLibDir "lib" CACHE STRING
+    "Install directory for plugin libraries PREFIX/PLUGIN_LIB_DIR/{lv2,vst}")
 
 # Now, handle the incoming settings and set define flags/variables based
 # on this
diff --git a/src/Plugin/AlienWah/CMakeLists.txt b/src/Plugin/AlienWah/CMakeLists.txt
index 5aed698f..40e67c6a 100644
--- a/src/Plugin/AlienWah/CMakeLists.txt
+++ b/src/Plugin/AlienWah/CMakeLists.txt
@@ -17,8 +17,8 @@ set_target_properties(ZynAlienWah_vst PROPERTIES PREFIX "")
 target_link_libraries(ZynAlienWah_lv2 zynaddsubfx_core ${OS_LIBRARIES})
 target_link_libraries(ZynAlienWah_vst zynaddsubfx_core ${OS_LIBRARIES})
 
-install(TARGETS ZynAlienWah_lv2 LIBRARY DESTINATION lib/lv2/ZynAlienWah.lv2/)
-install(TARGETS ZynAlienWah_vst LIBRARY DESTINATION lib/vst/)
+install(TARGETS ZynAlienWah_lv2 LIBRARY DESTINATION ${PluginLibDir}/lv2/ZynAlienWah.lv2/)
+install(TARGETS ZynAlienWah_vst LIBRARY DESTINATION ${PluginLibDir}/vst/)
 
 add_custom_command(TARGET ZynAlienWah_lv2 POST_BUILD
     COMMAND lv2-ttl-generator $<TARGET_FILE:ZynAlienWah_lv2>
@@ -28,4 +28,4 @@ install(FILES
 	${CMAKE_CURRENT_BINARY_DIR}/lv2/manifest.ttl
 	${CMAKE_CURRENT_BINARY_DIR}/lv2/presets.ttl
 	${CMAKE_CURRENT_BINARY_DIR}/lv2/ZynAlienWah.ttl
-	DESTINATION lib/lv2/ZynAlienWah.lv2/)
+	DESTINATION ${PluginLibDir}/lv2/ZynAlienWah.lv2/)
diff --git a/src/Plugin/Chorus/CMakeLists.txt b/src/Plugin/Chorus/CMakeLists.txt
index 17b4da06..ba538b9c 100644
--- a/src/Plugin/Chorus/CMakeLists.txt
+++ b/src/Plugin/Chorus/CMakeLists.txt
@@ -17,8 +17,8 @@ set_target_properties(ZynChorus_vst PROPERTIES PREFIX "")
 target_link_libraries(ZynChorus_lv2 zynaddsubfx_core ${OS_LIBRARIES})
 target_link_libraries(ZynChorus_vst zynaddsubfx_core ${OS_LIBRARIES})
 
-install(TARGETS ZynChorus_lv2 LIBRARY DESTINATION lib/lv2/ZynChorus.lv2/)
-install(TARGETS ZynChorus_vst LIBRARY DESTINATION lib/vst/)
+install(TARGETS ZynChorus_lv2 LIBRARY DESTINATION ${PluginLibDir}/lv2/ZynChorus.lv2/)
+install(TARGETS ZynChorus_vst LIBRARY DESTINATION ${PluginLibDir}/vst/)
 
 add_custom_command(TARGET ZynChorus_lv2 POST_BUILD
     COMMAND lv2-ttl-generator $<TARGET_FILE:ZynChorus_lv2>
@@ -28,4 +28,4 @@ install(FILES
 	${CMAKE_CURRENT_BINARY_DIR}/lv2/manifest.ttl
 	${CMAKE_CURRENT_BINARY_DIR}/lv2/presets.ttl
 	${CMAKE_CURRENT_BINARY_DIR}/lv2/ZynChorus.ttl
-	DESTINATION lib/lv2/ZynChorus.lv2/)
+    DESTINATION ${PluginLibDir}/lv2/ZynChorus.lv2/)
diff --git a/src/Plugin/Distortion/CMakeLists.txt b/src/Plugin/Distortion/CMakeLists.txt
index 1546b3e5..cfa9fd24 100644
--- a/src/Plugin/Distortion/CMakeLists.txt
+++ b/src/Plugin/Distortion/CMakeLists.txt
@@ -17,8 +17,8 @@ set_target_properties(ZynDistortion_vst PROPERTIES PREFIX "")
 target_link_libraries(ZynDistortion_lv2 zynaddsubfx_core ${OS_LIBRARIES})
 target_link_libraries(ZynDistortion_vst zynaddsubfx_core ${OS_LIBRARIES})
 
-install(TARGETS ZynDistortion_lv2 LIBRARY DESTINATION lib/lv2/ZynDistortion.lv2/)
-install(TARGETS ZynDistortion_vst LIBRARY DESTINATION lib/vst/)
+install(TARGETS ZynDistortion_lv2 LIBRARY DESTINATION ${PluginLibDir}/lv2/ZynDistortion.lv2/)
+install(TARGETS ZynDistortion_vst LIBRARY DESTINATION ${PluginLibDir}/vst/)
 
 add_custom_command(TARGET ZynDistortion_lv2 POST_BUILD
     COMMAND lv2-ttl-generator $<TARGET_FILE:ZynDistortion_lv2>
@@ -28,4 +28,4 @@ install(FILES
 	${CMAKE_CURRENT_BINARY_DIR}/lv2/manifest.ttl
 	${CMAKE_CURRENT_BINARY_DIR}/lv2/presets.ttl
 	${CMAKE_CURRENT_BINARY_DIR}/lv2/ZynDistortion.ttl
-	DESTINATION lib/lv2/ZynDistortion.lv2/)
+    DESTINATION ${PluginLibDir}/lv2/ZynDistortion.lv2/)
diff --git a/src/Plugin/DynamicFilter/CMakeLists.txt b/src/Plugin/DynamicFilter/CMakeLists.txt
index 47aed84c..e3e89f52 100644
--- a/src/Plugin/DynamicFilter/CMakeLists.txt
+++ b/src/Plugin/DynamicFilter/CMakeLists.txt
@@ -17,8 +17,8 @@ set_target_properties(ZynDynamicFilter_vst PROPERTIES PREFIX "")
 target_link_libraries(ZynDynamicFilter_lv2 zynaddsubfx_core ${OS_LIBRARIES})
 target_link_libraries(ZynDynamicFilter_vst zynaddsubfx_core ${OS_LIBRARIES})
 
-install(TARGETS ZynDynamicFilter_lv2 LIBRARY DESTINATION lib/lv2/ZynDynamicFilter.lv2/)
-install(TARGETS ZynDynamicFilter_vst LIBRARY DESTINATION lib/vst/)
+install(TARGETS ZynDynamicFilter_lv2 LIBRARY DESTINATION ${PluginLibDir}/lv2/ZynDynamicFilter.lv2/)
+install(TARGETS ZynDynamicFilter_vst LIBRARY DESTINATION ${PluginLibDir}/vst/)
 
 add_custom_command(TARGET ZynDynamicFilter_lv2 POST_BUILD
     COMMAND lv2-ttl-generator $<TARGET_FILE:ZynDynamicFilter_lv2>
@@ -28,4 +28,4 @@ install(FILES
 	${CMAKE_CURRENT_BINARY_DIR}/lv2/manifest.ttl
 	${CMAKE_CURRENT_BINARY_DIR}/lv2/presets.ttl
 	${CMAKE_CURRENT_BINARY_DIR}/lv2/ZynDynamicFilter.ttl
-	DESTINATION lib/lv2/ZynDynamicFilter.lv2/)
+    DESTINATION ${PluginLibDir}/lv2/ZynDynamicFilter.lv2/)
diff --git a/src/Plugin/Echo/CMakeLists.txt b/src/Plugin/Echo/CMakeLists.txt
index d93b5541..fe1ad015 100644
--- a/src/Plugin/Echo/CMakeLists.txt
+++ b/src/Plugin/Echo/CMakeLists.txt
@@ -17,8 +17,8 @@ set_target_properties(ZynEcho_vst PROPERTIES PREFIX "")
 target_link_libraries(ZynEcho_lv2 zynaddsubfx_core ${OS_LIBRARIES})
 target_link_libraries(ZynEcho_vst zynaddsubfx_core ${OS_LIBRARIES})
 
-install(TARGETS ZynEcho_lv2 LIBRARY DESTINATION lib/lv2/ZynEcho.lv2/)
-install(TARGETS ZynEcho_vst LIBRARY DESTINATION lib/vst/)
+install(TARGETS ZynEcho_lv2 LIBRARY DESTINATION ${PluginLibDir}/lv2/ZynEcho.lv2/)
+install(TARGETS ZynEcho_vst LIBRARY DESTINATION ${PluginLibDir}/vst/)
 
 add_custom_command(TARGET ZynEcho_lv2 POST_BUILD
     COMMAND lv2-ttl-generator $<TARGET_FILE:ZynEcho_lv2>
@@ -28,4 +28,4 @@ install(FILES
 	${CMAKE_CURRENT_BINARY_DIR}/lv2/manifest.ttl
 	${CMAKE_CURRENT_BINARY_DIR}/lv2/presets.ttl
 	${CMAKE_CURRENT_BINARY_DIR}/lv2/ZynEcho.ttl
-	DESTINATION lib/lv2/ZynEcho.lv2/)
+    DESTINATION ${PluginLibDir}/lv2/ZynEcho.lv2/)
diff --git a/src/Plugin/Phaser/CMakeLists.txt b/src/Plugin/Phaser/CMakeLists.txt
index eaa2f077..99262bb2 100644
--- a/src/Plugin/Phaser/CMakeLists.txt
+++ b/src/Plugin/Phaser/CMakeLists.txt
@@ -17,8 +17,8 @@ set_target_properties(ZynPhaser_vst PROPERTIES PREFIX "")
 target_link_libraries(ZynPhaser_lv2 zynaddsubfx_core ${OS_LIBRARIES})
 target_link_libraries(ZynPhaser_vst zynaddsubfx_core ${OS_LIBRARIES})
 
-install(TARGETS ZynPhaser_lv2 LIBRARY DESTINATION lib/lv2/ZynPhaser.lv2/)
-install(TARGETS ZynPhaser_vst LIBRARY DESTINATION lib/vst/)
+install(TARGETS ZynPhaser_lv2 LIBRARY DESTINATION ${PluginLibDir}/lv2/ZynPhaser.lv2/)
+install(TARGETS ZynPhaser_vst LIBRARY DESTINATION ${PluginLibDir}/vst/)
 
 add_custom_command(TARGET ZynPhaser_lv2 POST_BUILD
     COMMAND lv2-ttl-generator $<TARGET_FILE:ZynPhaser_lv2>
@@ -28,4 +28,4 @@ install(FILES
 	${CMAKE_CURRENT_BINARY_DIR}/lv2/manifest.ttl
 	${CMAKE_CURRENT_BINARY_DIR}/lv2/presets.ttl
 	${CMAKE_CURRENT_BINARY_DIR}/lv2/ZynPhaser.ttl
-	DESTINATION lib/lv2/ZynPhaser.lv2/)
+    DESTINATION ${PluginLibDir}/lv2/ZynPhaser.lv2/)
diff --git a/src/Plugin/Reverb/CMakeLists.txt b/src/Plugin/Reverb/CMakeLists.txt
index 58aa25e7..a38f5b61 100644
--- a/src/Plugin/Reverb/CMakeLists.txt
+++ b/src/Plugin/Reverb/CMakeLists.txt
@@ -17,8 +17,8 @@ set_target_properties(ZynReverb_vst PROPERTIES PREFIX "")
 target_link_libraries(ZynReverb_lv2 zynaddsubfx_core ${OS_LIBRARIES})
 target_link_libraries(ZynReverb_vst zynaddsubfx_core ${OS_LIBRARIES})
 
-install(TARGETS ZynReverb_lv2 LIBRARY DESTINATION lib/lv2/ZynReverb.lv2/)
-install(TARGETS ZynReverb_vst LIBRARY DESTINATION lib/vst/)
+install(TARGETS ZynReverb_lv2 LIBRARY DESTINATION ${PluginLibDir}/lv2/ZynReverb.lv2/)
+install(TARGETS ZynReverb_vst LIBRARY DESTINATION ${PluginLibDir}/vst/)
 
 add_custom_command(TARGET ZynReverb_lv2 POST_BUILD
     COMMAND lv2-ttl-generator $<TARGET_FILE:ZynReverb_lv2>
@@ -28,4 +28,4 @@ install(FILES
 	${CMAKE_CURRENT_BINARY_DIR}/lv2/manifest.ttl
 	${CMAKE_CURRENT_BINARY_DIR}/lv2/presets.ttl
 	${CMAKE_CURRENT_BINARY_DIR}/lv2/ZynReverb.ttl
-	DESTINATION lib/lv2/ZynReverb.lv2/)
+    DESTINATION ${PluginLibDir}/lv2/ZynReverb.lv2/)
diff --git a/src/Plugin/ZynAddSubFX/CMakeLists.txt b/src/Plugin/ZynAddSubFX/CMakeLists.txt
index 111bdb3e..247e8ead 100644
--- a/src/Plugin/ZynAddSubFX/CMakeLists.txt
+++ b/src/Plugin/ZynAddSubFX/CMakeLists.txt
@@ -52,8 +52,8 @@ set_target_properties(ZynAddSubFX_vst PROPERTIES PREFIX "")
 target_link_libraries(ZynAddSubFX_lv2 zynaddsubfx_core ${OS_LIBRARIES} ${LIBLO_LIBRARIES})
 target_link_libraries(ZynAddSubFX_vst zynaddsubfx_core ${OS_LIBRARIES} ${LIBLO_LIBRARIES})
 
-install(TARGETS ZynAddSubFX_lv2 LIBRARY DESTINATION lib/lv2/ZynAddSubFX.lv2/)
-install(TARGETS ZynAddSubFX_vst LIBRARY DESTINATION lib/vst/)
+install(TARGETS ZynAddSubFX_lv2 LIBRARY DESTINATION ${PluginLibDir}/lv2/ZynAddSubFX.lv2/)
+install(TARGETS ZynAddSubFX_vst LIBRARY DESTINATION ${PluginLibDir}/vst/)
 
 add_custom_command(TARGET ZynAddSubFX_lv2 POST_BUILD
     COMMAND lv2-ttl-generator $<TARGET_FILE:ZynAddSubFX_lv2>
@@ -63,7 +63,7 @@ install(FILES
 	${CMAKE_CURRENT_BINARY_DIR}/lv2/manifest.ttl
 	${CMAKE_CURRENT_BINARY_DIR}/lv2/presets.ttl
 	${CMAKE_CURRENT_BINARY_DIR}/lv2/ZynAddSubFX.ttl
-	DESTINATION lib/lv2/ZynAddSubFX.lv2/)
+    DESTINATION ${PluginLibDir}/lv2/ZynAddSubFX.lv2/)
 
 if(NtkGui)
 set_target_properties(ZynAddSubFX_lv2_ui PROPERTIES COMPILE_DEFINITIONS "DISTRHO_PLUGIN_TARGET_LV2")
@@ -71,9 +71,9 @@ set_target_properties(ZynAddSubFX_lv2_ui PROPERTIES LIBRARY_OUTPUT_DIRECTORY "lv
 set_target_properties(ZynAddSubFX_lv2_ui PROPERTIES OUTPUT_NAME "ZynAddSubFX_ui")
 set_target_properties(ZynAddSubFX_lv2_ui PROPERTIES PREFIX "")
 
-install(TARGETS ZynAddSubFX_lv2_ui LIBRARY DESTINATION lib/lv2/ZynAddSubFX.lv2/)
+install(TARGETS ZynAddSubFX_lv2_ui LIBRARY DESTINATION ${PluginLibDir}/lv2/ZynAddSubFX.lv2/)
 
 install(FILES
         ${CMAKE_CURRENT_BINARY_DIR}/lv2/ZynAddSubFX_ui.ttl
-        DESTINATION lib/lv2/ZynAddSubFX.lv2/)
+        DESTINATION ${PluginLibDir}/lv2/ZynAddSubFX.lv2/)
 endif()
-- 
2.47.0

